waku/incentivization/txid_proof.nim

```diff
@@ -5,25 +5,39 @@ import waku/incentivization/rpc
 const SimpleTransferGasUsed = Quantity(21000)
 const TxReceiptQueryTimeout = 3.seconds
 
+type EligibilityManager* = ref object
+  web3: Web3
+
+# Initialize the eligibilityManager with a web3 instance
+proc init*(
+    T: type EligibilityManager, ethClient: string
+): Future[EligibilityManager] {.async.} =
+  result = EligibilityManager(web3: await newWeb3(ethClient))
+  # TODO: handle error if web3 instance is not established
```

Comment:
Should we handle the potential error here, or wherever this procedure is called? From what I see in the existing codebase, calls to `newWeb3` are not wrapped in a `try` if called from within a test (and we only use eligibility-related logic in tests so far).